lib/class-wp-theme-json-gutenberg.php

```diff
@@ -1119,13 +1126,41 @@ public function merge( $incoming, $update_or_remove = 'remove' ) {
 			}
 			foreach ( $to_append as $path_to_append ) {
 				$path          = array_merge( $metadata['path'], $path_to_append );
-				$incoming_node = _wp_array_get( $incoming_data, $path, array() );
+				$incoming_node = _wp_array_get( $incoming_data, $path, null );
 				$existing_node = _wp_array_get( $existing_data, $path, array() );
 
 				if ( empty( $incoming_node ) && empty( $existing_node ) ) {
 					continue;
 				}
 
+				// If we are merging with something that has no values
+				// core origin should be  treated as if the values were present
+				// and we can remove the origin.
```

Comment:
I believe we no longer want to use `we` phrasing in inline docs...
```suggestion
				// If merging with something that has no values,
				// core origin should be treated as if the values were present
				// and the origin can be removed.
```